static/app/components/compactSelect/list.tsx

```diff
@@ -54,6 +61,17 @@ interface BaseListProps<Value extends React.Key>
    */
   label?: React.ReactNode;
   size?: FormSize;
+  /**
+   * Upper limit for the number of options to display in the menu at a time. Users can
+   * still find overflowing options by using the search box (if `searchable` is true).
+   * If used, make sure to hoist selected options to the top, otherwise they may be
+   * hidden from view.
+   */
+  sizeLimit?: number;
+  /**
+   * Message to be displayed when some options are hidden due to `sizeLimit`.
+   */
+  sizeLimitMessage?: string;
```

Comment:
I'm open to other names for this prop. Chose `sizeLimit` because there's a [native attribute, `size`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select#attributes), that when used on `select` elements determines the number of options that should be visible before scrolling, a _similar but not the same_ effect.

Reply:
I like that as a name!